backend/src/api_parsers/dblp_parser.py

```diff
@@ -0,0 +1,106 @@
+from requests.exceptions import HTTPError
+from src.api_handlers.dblp.handler import DblpHandler
+from src.api_handlers.core.exceptions import QuotaExceededException
+from src.api_parsers.models import Source, Publication, Author
+from src.api_parsers.exceptions import (
+    NoAuthorsException,
+    NoAffiliationException,
+    DBLPQuotaExceededException,
+)
+
+
+class DBLPParser:
+    def __init__(self, keywords: str):
+        self.keywords = keywords
+        self.handler = DblpHandler()
+
+    def get_authors(self) -> list[Author]:
+        authors: list[Author] = []
+        try:
+            pub_response = self.handler.get_publications(self.keywords)
+            for page in pub_response:
+                page_authors = self._parse_publications_page(page)
+                authors.extend(page_authors)
+        except DBLPQuotaExceededException as err:
+            authors.extend(err.authors)
+            return authors
+        except (HTTPError, QuotaExceededException):
+            raise NoAuthorsException(self.keywords)
+        return authors
+
+    def _parse_publications_page(self, page: dict) -> list[Author]:
+        authors: list[Author] = []
+        hits = page["result"]["hits"]["hit"]
+        for hit in hits:
+            try:
+                authors.extend(self._parse_hit_dict(hit))
+            except DBLPQuotaExceededException as err:
+                authors.extend(err.authors)
+                raise DBLPQuotaExceededException(authors)
+        return authors
+
+    def _parse_hit_dict(self, hit: dict) -> list[Author]:
+        authors: list[Author] = []
+        info = hit["info"]
+        pub_data = {
+            "title": info["title"],
+            "abstract": "",
+            "citations": 0,
+            "year": info["year"],
+            "source_api": Source.DBLP,
+        }
+        publication = Publication(**pub_data)
+        authors_list = info["authors"]["author"]
+        if type(authors_list) == dict:
+            authors_list = [authors_list]
+        for author in authors_list:
+            author_id = author["@pid"]
+            author_name = author["text"]
+            split_name = author_name.split()
+            first_name, last_name = split_name[0], " ".join(split_name[1:])
+            try:
```

Comment:
please make it into separate field, we might need that

Reply:
the full name is literally split into separate fields here

Reply:
sorry, my bad